neuraxle/pipeline.py

```diff
@@ -486,3 +490,222 @@ def should_resume(self, data_container: DataContainer) -> bool:
                 return True
 
         return False
+
+
+"""
+Idea for checkpoints : 
+
+    The streaming pipeline algorithm could go find the optional checkpoint step for each sub pipeline.
+    In the future, a ResumableStreamingPipeline that extends this class should exist to support checkpoints.
+    
+    The Barrier should ideally join the data so that the ram does not blow up (iterable, lazy loading, cache ??)
+    Maybe we can implement a LazyLoadingDataContainer/CachedDataContainer class or something that could be returned.
+    
+    pipeline = Pipeline([
+        MiniBatchSequentialPipeline([
+
+            A(),
+            B(),
+            Barrier(joiner=Joiner()),
+            [Checkpoint()]
```

Comment:
I guess this wouldn't be in a `[  ]` list but that the `Checkpoint()` would be by itself, not like `[Checkpoint()]`.

Reply:
[] is for facultative